### PR TITLE
refactor: align file-input with others

### PR DESCRIPTION
**Pull request by HendrikThePendric**
- FileInput no longer includes the label
- FileInputField includes label, help and error
- Other inputs have a tabIndex property so we added that to FileInput(Field) too. However, since the visible element on the form is a button, the Button component needed to get a tabIndex
- We are not longer exporting name-spaced components, FileList, FileListItem and FileListPlaceholder are exported separately
- Stories were updated
- Small tweak to ESLint config so it would allow me to commit

### CODE REVIEW

File: src/FileInput.js [hunk 1]
```diff
@@ -22,32 +17,26 @@ class FileInput extends PureComponent {
 
     onChange = () => {
         this.props.onChange(this.ref.current.files)
-        // reset the file input so it won't prevent on-change
```
[the remark this answers is not in the document]
Comment (HendrikThePendric) [reply src/FileInput.js]: I've restored the comment. It made sense to have it there

**Comment by HendrikThePendric** [comment]
@Mohammer5 I've added the jsDoc comments for all new components now. Ready for re-review...

File: .eslintrc.js [hunk 2]
```diff
@@ -3,4 +3,9 @@ const { config } = require('@dhis2/cli-style')
 module.exports = {
     parser: 'babel-eslint',
     extends: ['eslint:recommended', 'plugin:react/recommended', config.eslint],
+    settings: {
+        react: {
+            version: 'detect',
```
Comment (varl) [inline .eslintrc.js]: This could be 16.3 for now.

Comment (HendrikThePendric) [reply .eslintrc.js]: 👍 https://github.com/dhis2/ui-core/pull/443

**Pull request merged**